tests/main/snapd-sigterm/task.yaml

```diff
@@ -21,7 +21,12 @@ execute: |
     TEST_TIME0="$(date +'%s')"
     systemctl stop snapd.service
 
-    retry -n 10 sh -c 'systemctl status snapd.service | MATCH "inactive"'
+    # The systemctl command waits for the operation to complete, but just to be
+    # extra safe check that it's either inactive or has just been restarted. We
+    # don't want to stop the socket itself, as that might hide the issue we
+    # want to test.
+    retry -n 100 --wait 0.1 sh -c 'systemctl status snapd.service | MATCH "inactive|activating"'
```

Comment:
Hmm the problem here was that we didn't know why snapd is being restarted? Perhaps it's nc that's poking the socket, and thus making systemd activate the service?

I wonder it it'd be enough to have a simple client in python just to simulate the behavior:
```
python3 -c 'import socket; s=socket.socket(socket.SOCK_STREAM,socket.AF_UNIX);s.connect('/run/snapd.socket');s.write("GET /v2/apps HTTP/1.1\r\n\r\n")'
```